crates/networking/p2p/rlpx/eth.rs

```diff
@@ -109,3 +109,314 @@ impl RLPxMessage for StatusMessage {
         })
     }
 }
+
+// https://github.com/ethereum/devp2p/blob/master/caps/eth.md#getblockbodies-0x05
+#[derive(Debug)]
+pub(crate) struct GetBlockBodies {
+    // id is a u64 chosen by the requesting peer, the responding peer must mirror the value for the response
+    // https://github.com/ethereum/devp2p/blob/master/caps/eth.md#protocol-messages
+    id: u64,
+    block_hashes: Vec<BlockHash>,
+}
+
+impl GetBlockBodies {
+    pub fn build_from(id: u64, block_hashes: Vec<BlockHash>) -> Self {
+        Self { block_hashes, id }
+    }
+}
+
+impl RLPxMessage for GetBlockBodies {
+    fn encode(&self, buf: &mut dyn BufMut) {
+        let mut encoded_data = vec![];
+        Encoder::new(&mut encoded_data)
+            .encode_field(&self.id)
+            .encode_field(&self.block_hashes)
+            .finish();
+
+        let mut snappy_encoder = SnappyEncoder::new();
+        let mut msg_data = vec![0; max_compress_len(encoded_data.len()) + 1];
+
+        let compressed_size = snappy_encoder
+            .compress(&encoded_data, &mut msg_data)
+            .unwrap();
+
+        msg_data.truncate(compressed_size);
+
+        buf.put_slice(&msg_data);
+    }
+
+    fn decode(msg_data: &[u8]) -> Result<Self, RLPDecodeError> {
+        let mut snappy_decoder = SnappyDecoder::new();
+        let decompressed_data = snappy_decoder
+            .decompress_vec(msg_data)
+            .map_err(|err| RLPDecodeError::Custom(err.to_string()))?;
+        let decoder = Decoder::new(&decompressed_data)?;
+        let (id, decoder): (u64, _) = decoder.decode_field("request-id")?;
+        let (block_hashes, _): (Vec<BlockHash>, _) = decoder.decode_field("blockHashes")?;
+
+        Ok(Self { block_hashes, id })
```

Comment:
Maybe use the constructor here (to remove the CI warning on:
```error: associated function `build_from` is never used```
(or remove the constructor at all)